lib/media/track/localmediatrack.js

```diff
@@ -321,22 +322,49 @@ function restartWhenInadvertentlyStopped(localMediaTrack) {
       // NOTE(mmalavalli): If the MediaStreamTrack ends before the DOM is visible,
       // then this makes sure that visibility callback for phase 2 is called only
       // after the MediaStreamTrack is re-acquired.
-      return trackChangeInProgress && trackChangeInProgress.promise;
+      const promise = (trackChangeInProgress && trackChangeInProgress.promise) || Promise.resolve();
+      return promise.finally(() => localMediaRestartDeferreds.resolveDeferred(kind));
     });
   }
 
+  function onMute() {
+    const { _log: log, kind } = localMediaTrack;
+    log.info('Muted');
+    log.debug('LocalMediaTrack:', localMediaTrack);
+
+    // NOTE(mmalavalli): When a LocalMediaTrack is muted without the app being
+    // backgrounded, and the inadvertently paused elements are played before it
+    // is restarted, it never gets unmuted due to the WebKit Bug 213853. Hence,
```

Comment:
@manjeshbhargav - when the element gets muted because of phone call, does it ever get unmuted? what ensures that we will call `maybeRestart` ?

Reply:
@makarandp0 If you play the paused elements before the audio track is ended/unmuted, then the audio track does not get ended/unmuted, so `maybeRestart` is never called. So, we wait for unmuted/ended to fire, and then play the paused elements.

Reply:
and I assume we do get `unmuted` in those cases? LGTM

Reply:
@makarandp0 We get the `ended` event.